pupil_src/shared_modules/fixation_detector.py

```diff
@@ -158,41 +142,47 @@ def gaze_dispersion(capture, gaze_subset, use_pupil=True):
 
         # undistort onto 3d plane
         vectors = capture.intrinsics.unprojectPoints(locations)
+    else:
+        raise ValueError(f"Unknown method '{method}'")
 
     dist = vector_dispersion(vectors)
-    return dist, method, base_data
+    return dist
 
 
 def detect_fixations(
     capture, gaze_data, max_dispersion, min_duration, max_duration, min_data_confidence
 ):
     yield "Detecting fixations...", ()
-    gaze_data = [
+    gaze_data = (
         fm.Serialized_Dict(msgpack_bytes=serialized) for serialized in gaze_data
+    )
+    gaze_data = [
+        datum for datum in gaze_data if datum["confidence"] > min_data_confidence
     ]
     if not gaze_data:
         logger.warning("No data available to find fixations")
-        return "Fixation detection complete", ()
+        return "Fixation detection failed", ()
 
-    use_pupil = can_use_3d_gaze_mapping(gaze_data)
-    logger.info(
-        "Starting fixation detection using {} data...".format(
-            "3d" if use_pupil else "2d"
-        )
+    method = (
+        FixationDetectionMethod.GAZE_3D
+        if can_use_3d_gaze_mapping(gaze_data)
+        else FixationDetectionMethod.GAZE_2D
     )
+    logger.info(f"Starting fixation detection using {method.value} data...")
     fixation_result = Fixation_Result_Factory()
 
     Q = deque()
     enum = deque(gaze_data)
```

Comment:
Actually since we are making this nice currently, can we rename the deques (especially the `enum`)?
`Q` is more like `working_queue`
and `enum` should be something along the lines of `remaining_gaze_data`.